lib/api/funnel.js

```diff
@@ -368,9 +369,29 @@ class Funnel {
    * @returns {Promise<Request>}
    */
   async checkRights (request) {
+   
+    let skipTokenVerification = false;
+    // When the Support of Cookie as Authentication Token is enabled we check if an auth token cookie is present
+    // When a request is made with CookieOnly set to true
+    // We try to use the auth token cookie if present as auth token
+    // otherwise check for auth token as input
+    if ( global.kuzzle.config.security.supportCookieAuthentication
+      && request.input.args.cookieOnly
+      && request.input.headers
+      && request.input.headers.cookie
```

Comment:
You should use `has(object, property)` from `lib/util/safeObject.js`, it's safer.

```suggestion
      && has(request.input.headers, 'cookie')
```